lib/src/main/java/io/ably/lib/util/AgentHeaderCreator.java

```diff
@@ -0,0 +1,61 @@
+package io.ably.lib.util;
+
+import android.os.Build;
```

Comment:
That worries me a bit because I'm not sure if that will cause troubles for the `ably-java` variant which doesn't have any android dependencies, right? :thinking:

Reply:
Yes, this is probably a problem. This project has a _unique_ structure, so we need to tread carefully. There is an Android specific folder so you may need to create an API to surface additional headers for whatever platform we're running on.

Reply:
I've created a solution that works but I'm not very satisfied with how it looks like https://github.com/ably/ably-java/pull/671/commits/2561db3064b8e97b6a9454070038eee1bef6db3e but due to the project structure I haven't had any better idea so if you'll have any suggestions on how to improve it that would be awesome :smile:

Reply:
Yeah, I agree. As discussed at more length with you via Slack DMs, the mutable static field isn't a great solution. We need to move this up to somewhere it can be governed by the compiler. Back to the drawing board. ✏️

Reply:
I've changed it so now each of the platforms (java, android) provide their own `PlatformAgentProvider` which returns the correct agent. This shouldn't affect the way of using the SDK and wee will avoid using that mutable static field :wink: https://github.com/ably/ably-java/pull/671/commits/08c9e6281a647e981516dd23ae75452cb23da29c
@paddybyers could you take a look at the above commit in terms of the SDKs architecture? Does it look alright to you?